sql/core/src/test/scala/org/apache/spark/sql/DataFrameFunctionsSuite.scala

```diff
@@ -209,6 +209,18 @@ class DataFrameFunctionsSuite extends QueryTest with SharedSQLContext {
       Row(2743272264L, 2180413220L))
   }
 
+  test("misc data_type function") {
+    val df = Seq(("a", false)).toDF("a", "b")
+
+    checkAnswer(
+      df.select(data_type($"a"), data_type($"b")),
```

Comment:
I think we can easily get the types via

```scala
val df = Seq(("a", false)).toDF("a", "b")
df.schema.map(_.dataType.simpleString)
```

I just wonder in which case we need these type per each row in the dataframe.